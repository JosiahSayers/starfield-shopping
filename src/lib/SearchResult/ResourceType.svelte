<script lang="ts">
  export let type: string;
</script>

<span class="badge badge-neutral m-1">{type}</span>
