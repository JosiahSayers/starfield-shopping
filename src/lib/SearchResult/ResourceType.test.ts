import { describe, afterEach, it, expect } from 'vitest';
import { cleanup, render } from '@testing-library/svelte'
import ResourceType from './ResourceType.svelte'

describe('ResourcePrice', () => {
  afterEach(() => cleanup())

  it('mounts', () => {
    const { container } = render(ResourceType, { type: '' });
    expect(container).toBeTruthy();
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders the price', () => {
    const { container } = render(ResourceType, { type: 'test-type' });
    expect(container).toBeTruthy();
    expect(container.querySelector('.badge')?.textContent).toBe('test-type');
  });
});
