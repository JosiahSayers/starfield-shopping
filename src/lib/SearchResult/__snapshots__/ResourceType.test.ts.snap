// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ResourcePrice > mounts 1`] = `"<div><span class=\\"badge badge-neutral m-1\\"></span></div>"`;
